src/libs/form/components/NumberSlider.tsx

```diff
@@ -56,43 +77,73 @@ export function NumberSlider({
   );
 
   const handleSliderFocus = (event: React.FocusEvent<HTMLDivElement>) => {
-    setFocused(true);
+    setState((current) => ({ ...current, focused: true }));
     sliderProps?.onFocus?.(event);
   };
 
   const handleSliderBlur = (event: React.FocusEvent<HTMLDivElement>) => {
-    setFocused(false);
+    setState((current) => ({ ...current, focused: false }));
     sliderProps?.onBlur?.(event);
   };
 
   const handleInputFocus = (event: React.FocusEvent<HTMLInputElement>) => {
-    setFocused(true);
+    setState((current) => ({ ...current, focused: true }));
     numberProps?.onFocus?.(event);
   };
 
   const handleInputBlur = (event: React.FocusEvent<HTMLInputElement>) => {
-    setFocused(false);
+    setState((current) => ({ ...current, focused: false }));
     numberProps?.onBlur?.(event);
   };
 
   useEffect(() => {
-    if (!focused) {
-      setValue(value);
+    if (!state.focused) {
+      setState((current) => ({ ...current, _value: value }));
     }
-  }, [value, precision]);
+  }, [value, precision, state.focused]);
 
   useEffect(() => {
-    if (!changeEndValue) return;
-    onChange?.(changeEndValue);
-  }, [changeEndValue]);
+    if (!state.changeEndValue) return;
+    onChange?.(state.changeEndValue);
+  }, [state.changeEndValue]);
 
   useEffect(() => {
     if (!numberRef.current) return;
-    setComputedWidth(getComputedWidth(numberRef.current, min, max, precision));
+    setState((current) => ({
+      ...current,
+      // Just to keep ts happy :shrug:
+      computedWidth: numberRef.current
```

Comment:
There's a check for `!numberRef.current` above... The heck is TS complaining about 👀

Reply:
I know, I know 😬 but it complains anyways